javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,16 @@
+package java.util.concurrent.locks
+
+class ReentrantLock(private[this] val fair: Boolean) {
+
+  private var locked = false
+
+  def this() = this(false)
+
+  def lock(): Unit = locked = true
+
+  def unlock(): Unit = locked = false
```

Comment:
`unlock()` must throw `IllegalMonitorException` if the lock is not currently locked.